functions/pupil-status/pupil-status.service.js

```diff
@@ -69,6 +71,36 @@ async function changePupilState (pupilId, targetStatusCode) {
   return sqlService.modify(sql, params)
 }
 
+/**
+ * Filter for live checks and make a request for the pupil-status to be updated for multiple pupils on v2
+ * @param {Object} logger
+ * @param {String} logPrefix
+ * @param {[{checkId: <number>, pupilId: <number>, checkCode: <string>, isLiveCheck: <boolean>}]} checkData - must contain `checkId`, `pupilId`, `checkCode` and `isLiveCheck` props
+ * @return {Promise<*|Promise<*>>}
+ */
+async function updatePupilStatusForLiveChecksV2 (logger, logPrefix, checkData) {
+  if (!checkData || !Array.isArray(checkData)) {
+    logger.error(`${logPrefix}: updatePupilStatusV2(): ERROR: check data provided must be an array`)
+    return
+  }
+  logger.info(`${logPrefix}: updatePupilStatusV2(): got ${checkData.length} pupils`)
+  // Batch the async messages up for live checks only, to limit max concurrency
+  const batches = R.splitEvery(100, R.filter(c => c.isLiveCheck, checkData))
+  checkData = null
+  batches.forEach(async (checks, batchNumber) => {
+    try {
+      await azureStorageHelper.addMessageToQueue('pupil-status', {
+        version: 2,
+        messages: checks
```

Comment:
This code does not create well-formed pupil-status:v2 messages!  Whilst it will work, the extra properties `checkId` and `isLiveCheck` have no place in the message.

Reply:
@jon-shipley fair point, I changed this to provide a list of check records with the props `pupilId` and `checkCode` which are utilised by the `v2` implementation.